src/main/java/org/jitsi/videobridge/AbstractEndpoint.java

```diff
@@ -318,13 +318,42 @@ public void describe(ColibriConferenceIQ.ChannelBundle channelBundle)
     public JSONObject getDebugState()
     {
         JSONObject debugState = new JSONObject();
+        debugState.put("receiverVideoConstraints", receiverVideoConstraintsMap);
+        debugState.put("maxReceiverVideoConstraints", maxReceiverVideoConstraints);
         debugState.put("displayName", displayName);
         debugState.put("expired", expired);
         debugState.put("statsId", statsId);
 
         return debugState;
     }
 
+    /**
+     * Computes and sets the {@link #maxReceiverVideoConstraints} from the
+     * specified video constraints.
+     *
+     * @param newVideoConstraints the map of receiver endpoint id -> video
+     * constraints that specifies who (which receiver endpoint) is viewing what
+     * (as determined by the video constraints) from this endpoint (which is the
+     * sender).
+     */
+    private void receiverVideoConstraintsChanged(
+        Collection<VideoConstraints> newVideoConstraints)
+    {
+        VideoConstraints oldReceiverMaxVideoConstraints
+            = this.maxReceiverVideoConstraints;
+
+        VideoConstraints newReceiverMaxVideoConstraints = newVideoConstraints
+            .stream()
+            .max(Comparator.comparingInt(VideoConstraints::getIdealHeight))
+            .orElse(defaultMaxReceiverVideoConstraints);
+
+        if (!newReceiverMaxVideoConstraints.equals(oldReceiverMaxVideoConstraints))
```

Comment:
Same as before (`equals` takes into account more than `idealHeight`). I though that's why we planned to do the `VideoConstraints` refactor as a first step?

Reply:
> Same as before (equals takes into account more than idealHeight).

I can certainly change this line to compare the ideal height only, but 1) in practice this won't fix nor change anything, and 2) when we add an `idealFps` field we'll have to to remember to come back here and augment the comparison. 

Eventually we will separate the video constraints (ideal fps, ideal resolution) from the video allocation policy (preferred fps, preferred resolution) and this equality comparison will become less awkward.

Let me know if you wish me to elaborate on 1) and/or if you feel this needs to compare the ideal height only, regardless. 

> I though that's why we planned to do the VideoConstraints refactor as a first step?

Yes.

Reply:
Makes sense for now, I just expected the other step first.

Reply:
I agree it would be nice to have it as a first step, but @bbaldino rejected my previous suggestion, so I'd leave this as a future step when we're ready to tackle it.